src/test_l3_dsCompositeIn.c

```diff
@@ -0,0 +1,536 @@
+    for(int i = 0 ; i < dsCOMPOSITE_IN_PORT_MAX ; i++) 
+    {
+        UT_LOG_INFO("Result dsCompositeInGetStatus(OUT:inputstatus[isPresented:%s, isPortConnected:%s, activeport:%s])",
+                UT_Control_GetMapString(bool_mapTable, inputstatus.isPresented),
+                UT_Control_GetMapString(bool_mapTable, inputstatus.isPortConnected[i]),
+                UT_Control_GetMapString(dsCompositeInPortMappingTable, inputstatus.activePort));
+    }
+    ASSERT(ret == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to select port.
+*
+* This test function selects CompositeIn port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 004@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [dsCompositeIn_L3_Low-Level_TestSpecification.md](../docs/pages/ds-compositeIn_L3_Low-Level_TestSpecification.md)
+*/
+void test_l3_CompositeIn_select_port(void)
+{
+    gTestID = 4;
+    UT_LOG_INFO("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    dsCompositeInPort_t port = dsCOMPOSITE_IN_PORT_MAX;
+    int32_t select; 
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsCompositeIn/composite_input_configurations/number_of_ports");
+
+    UT_LOG_INFO("----------------------------------------------------------");
+    UT_LOG_INFO("Available CompositeIn Ports");
+    UT_LOG_INFO("----------------------------------------------------------");
+    UT_LOG_INFO("\t#  %-20s","CompositeIn Port");
+    for(int32_t i = 0; i < numInputPorts; i++)
+    {
+        UT_LOG_INFO("\t%d.  %-20s", i, UT_Control_GetMapString(dsCompositeInPortMappingTable, i));
+    }
+    UT_LOG_INFO("----------------------------------------------------------");
+    UT_LOG_INFO("Enter the port to select: ");
+    readInput(&select);
+
+    if(select <= 0 || select > numInputPorts) 
+    {
+       UT_LOG_ERROR("\nInvalid port selected\n");
+       goto exit;
+    }
+
+    port = (dsCompositeInPort_t)select;
+
+    UT_LOG_INFO("Calling dsCompositeInSelectPort(IN:port[%d])",
+                UT_Control_GetMapString(dsCompositeInPortMappingTable, port));
+    ret = dsCompositeInSelectPort(port);
+    UT_LOG_INFO("Result dsCompositeInSelectPort(IN:port[%d] dsError_t:[%s])",
+                UT_Control_GetMapString(dsCompositeInPortMappingTable, port), 
+                UT_Control_GetMapString(dsError_mapTable, ret));
+    ASSERT(ret == dsERR_NONE);
+
+    exit:
+        UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to scale video.
+*
+* This test function is to scale video of selected CompositeIn port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 005@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [dsCompositeIn_L3_Low-Level_TestSpecification.md](../docs/pages/ds-compositeIn_L3_Low-Level_TestSpecification.md)
+*/
+void test_l3_CompositeIn_scale_video(void)
```

Comment:
scale is applied on port right 
so ask user to select the port

Reply:
done

Reply:
ok fine